modules/core/src/hal_replacement.hpp

```diff
@@ -717,8 +717,12 @@ inline int hal_ni_gemm64fc(const double* src1, size_t src1_step, const double* s
 
 #if defined __GNUC__
 #  pragma GCC diagnostic pop
+#define EXPAND
 #elif defined _MSC_VER
 #  pragma warning( pop )
+#define EXPAND( x ) x
+#else
+#define EXPAND
```

Comment:
Macros with generic names should be avoided, especially in public headers (to prevent possible conflicts). For example, use `CV_` prefix.

Consider to reuse [this macro](https://github.com/opencv/opencv/blob/3.3.0/modules/core/include/opencv2/core/cvdef.h#L73).

Reply:
Yes its absolutely the correct design to reuse this one.